### ISSUE TITLE
Inbuilt support for content assist on typing like vscode

### ISSUE THREAD

**Issue by gayanper** [eclipse-platform/eclipse.platform.ui#888]
Today for JavaEditor we can achieve this my providing all the letters in the activation trigger. But the experience is not so good due to auto activation doesn't debounce the key events.  For example when you are trying continuously we don't want to content assist to popup and preselect wrong suggestions which the developer doesn't need.

The current activation logic can be extended to support debounce, We could introduce a configuration to enable this new activation feature for all valid keys with debounce. One simple implementation would be to change the current activation logic by resetting the last activation time when ever a key is pressed and reduce the delay to something like 100ms in this new mode.

**Comment by gayanper** [comment]
@mickaelistria  @vogella  Do you think this is a good idea, if so I can try out and push a PR

**Comment by mickaelistria** [comment]
Yes, it's a good idea that was tracked in https://bugs.eclipse.org/bugs/show_bug.cgi?id=101420 . I suggest you look into the comments and blockers in more details before going too deep. It's not as easy as it seems in the current state and we'll probably need preliminary fixes in JDT or other before this can be achieve smoothly.

**Comment by mickaelistria** [comment]
See also eclipse-platform/eclipse.platform.text#66 which enables it by default in generic editor. It's very nice to use!
